src/main/kotlin/edu/kit/compiler/semantic/visitor/MainMethodVerifier.kt

```diff
@@ -16,6 +16,20 @@ class MainMethodVerifier(val sourceFile: SourceFile) : AbstractVisitor() {
         // don't visit method blocks, that would be waste of time
     }
 
+    override fun visitMethodInvocationExpression(methodInvocationExpression: AstNode.Expression.MethodInvocationExpression) {
+        if (methodInvocationExpression.method.symbol.text == "main")
+            sourceFile.annotate(
+                AnnotationType.ERROR,
+                methodInvocationExpression.sourceRange,
+                "the main method cannot be invoked.",
+                listOf(
+                    SourceNote(methodInvocationExpression.sourceRange, "hint: don't call the main method.")
```

Comment:
Notes sind eig dafür gedacht, dass man noch weitere Code-Stellen anzeigen kann in der Fehlermeldung^^ Das hier aktuell führt dazu, dass die gleiche Code-Zeile 2x wiederholt wird, was eher nicht so geil is :D würde den hint einfach weglassen.

Außerdem wird der Code hier aktuell auch Programme ablehnen, die einfach nur eine Methode namens "main" haben, die aber keine Main-Methode ist^^ Glaube wir brauchen den check ehrlich gesagt nicht, weil das eh in der namens-analyse fehlschlägt^^